packages/react-query/src/QueryClientProvider.tsx

```diff
@@ -80,19 +49,7 @@ export const QueryClientProvider = ({
     }
   }, [client])
 
-  if (process.env.NODE_ENV !== 'production' && contextSharing) {
-    client
-      .getLogger()
-      .error(
-        `The contextSharing option has been deprecated and will be removed in the next major version`,
-      )
-  }
-
-  const Context = getQueryClientContext(context, contextSharing)
+  const Context = getQueryClientContext(context)
 
-  return (
-    <QueryClientSharingContext.Provider value={!context && contextSharing}>
-      <Context.Provider value={client}>{children}</Context.Provider>
-    </QueryClientSharingContext.Provider>
-  )
+  return <Context.Provider value={client}>{children}</Context.Provider>
```

Comment:
Maybe we can call the `Context` by it's name now that there's no name clashing?
```suggestion
  const QueryClientContext = getQueryClientContext(context)

  return <QueryClientContext.Provider value={client}>{children}</QueryClientContext.Provider>
```

Reply:
Yeah this will then match the solid-query context naming.

Reply:
Exactly!